provisioner/aws-manager.js

```diff
@@ -166,6 +271,254 @@ AwsManager.prototype._filterSpotRequests = function (spotReqs) {
+  // which fail we want stats on why they failed so we can later analyze this
+  differences.requests.forEach(function (request) {
+    if (request.State === 'active' && request.Status.Code === 'fulfilled') {
+      plotSpotFulfilment(request);
+    } else if (request.State === 'open') {
+      // Here we have those spot requests which are no longer unfulfilled but
+      // do not have an api state which reflects that
+      that.__awaitingSpotFulfilmentStatus.push({
+        id: request.SpotInstanceRequestId,
+        time: new Date(),
+        iterationCount: 0,
+      });
+    } else {
+      // I wonder if we should be reporting spot requests which failed because
+      // of bid amount here in their own special series for use in trying to
+      // figure out what to bid.  We could use refused spot bids as a lower
+      // limit for bids
+      that.reportSpotRequestsDied({
+        provisionerId: that.provisionerId,
+        region: request.Region,
+        az: request.LaunchSpecification.Placement.AvailabilityZone,
+        instanceType: request.LaunchSpecification.InstanceType,
+        workerType: request.WorkerType,
+        id: request.SpotInstanceRequestId,
+        time: dateForInflux(request.Status.UpdateTime),
+        bid: parseFloat(request.SpotPrice, 10),
+        state: request.State,
+        statusCode: request.Status.Code,
+        statusMsg: request.Status.Message,
+      });
+      debug('spot request %j did something else', request);
+    }
+  });
+
+  this.__awaitingSpotFulfilmentStatus = this.__awaitingSpotFulfilmentStatus.filter(function (requestAwaiting) {
+    var keepInTheList = true;
+
+    deadState.requests.forEach(function (requestMightHave) {
+      if (requestMightHave.SpotInstanceRequestId === requestAwaiting) {
+        if (requestMightHave.State === 'active' && requestMightHave.Status.Code === 'filfilled') {
+          plotSpotFulfilment(requestMightHave);
+          keepInTheList = false;
+        }
+      }
+    });
+
+    if (requestAwaiting.iterationCount++ > MAX_ITERATIONS_FOR_STATE_RESOLUTION) {
+      debug('dropping spot request on the floor');
+      keepInTheList = false;
+    }
+
+    return keepInTheList;
+  });
+
+  /*
+    for instances we need to see if there's a transition reason already and if not,
+    we need to store the instance metadata and check on the next iteration whether
+    that instance has an associated state transition reason.  For when there is a
+    spot price termination, we should also look for a spot request in the good and
+    dead ones to see what we bid on the thing.  We should also track what the time
+    was when we found the difference and we should only try for up to, say, 5 times
+    before we give up on trying to figure out why the instance terminated.
+
+    http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_StateReason.html
+  */
+
+  function plotInstanceDeath (instance, time) {
```

Comment:
Please start using ES6 arrow functions, there is no reason not to...

**example**
```js
let plotInstanceDeath = (instance, time) => {
  this.reportInstanceTerminated(....
};
```

Notice that you don't have to use the `that` thing, when you use arrow functions...